### PR TITLE
Issue 844

### PR DESCRIPTION
**Pull request by Dieterbe**
cc @shanson7

### CODE REVIEW

**Comment by Dieterbe** [comment]
because this is data critical code, I want reviews from at least 2 people.
note: I also checked #781 and the code should still cover all cases. except we don't need to check whether the rob has data because before where that check was, if it's non-nil, we flush it so it's always empty at that point.

File: mdata/aggmetric.go [hunk 1]
```diff
@@ -561,16 +561,45 @@ func (a *AggMetric) add(ts uint32, val float64) {
 	a.addAggregators(ts, val)
 }
 
+// collectable returns whether the AggMetric is garbage collectable
+// an Aggmetric is collectable based on two conditions:
+// * the AggMetric hasn't been written to in a configurable amount
```
Comment (replay) [inline mdata/aggmetric.go]: `amount... of time`?

Comment (Dieterbe) [reply mdata/aggmetric.go]: sure will fix

File: mdata/aggmetrics.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func (ms *AggMetrics) GC() {
 			ms.RLock()
 			a := ms.Metrics[key]
 			ms.RUnlock()
-			if stale := a.GC(chunkMinTs, metricMinTs); stale {
+			if stale := a.GC(now, chunkMinTs, metricMinTs); stale {
```
[the remark this answers is not in the document]
Comment (Dieterbe) [reply mdata/aggmetrics.go]: wow. blast from the past. did some digging. it was just how the very first version of GC was implemented.
da99d6d2d
i'll change this as well

**Pull request merged**